python/paddle/fluid/framework.py

```diff
@@ -1817,6 +1817,89 @@ def _sliceAndConcatVar(self, item, axis):
     def __getitem__(self, item):
         return _getitem_impl_(self, item)
 
+    def __setitem__(self, item, value):
+        inputs = {'Input': self}
+
+        # 1. Parse item
+        if not isinstance(item, tuple):
+            item = [item]
+
+        axes = []
+        starts = []
+        ends = []
+        max_integer = 2147483647
+        for dim, slice_item in enumerate(item):
+            if isinstance(slice_item, slice):
+                start = slice_item.start
+                end = slice_item.stop
+                step = slice_item.step
+
+                if start is None and end is None and step is None:
+                    continue
+
+                start = 0 if start is None else start
+                step = 1 if step is None else step
+
+                # TODO: support cases when step != 1
+                if step != 1:
+                    raise ValueError(
+                        "When assign a value to a paddle.Tensor, only support step is 1, "
+                        "but received step is {}.".format(step))
+                end = max_integer if end is None else end
+            else:
+                start = slice_item
+                end = slice_item + 1 if slice_item != -1 else max_integer
+            axes.append(dim)
+            starts.append(start)
+            ends.append(end)
+
+        attrs = {'axes': axes, 'starts': starts, 'ends': ends}
+
+        # 2. Parse value
+        dtype = self.dtype
+        attrs['dtype'] = dtype
+
+        #  2.1 value is an integer of float
+        if isinstance(value, (int, float)):
+            value = np.array([value])
+
+        #  2.2 value is a np.ndarray
+        if isinstance(value, np.ndarray):
+            shape = list(value.shape)
+            if dtype == core.VarDesc.VarType.BOOL:
+                value_name = "bool_values"
```

Comment:
I found you use attribute to make `setitem` op accept input for different numpy array types.

It is somehow not elegant code because of multiple input attributes and we add attribute for per-dtype.

Can we assign the ndarray to a Tensor and make the Tensor as the input of `setitem`? You may refer to `assign` API for initializing a Tensor with ndarray.

Reply:
> Can we assign the ndarray to a Tensor and make the Tensor as the input of setitem? You may refer to assign API for initializing a Tensor with ndarray.

No. And actually, `assign` API also uses attribute to  accept input for different numpy array types.